aii-core/src/main/perl/Shellfe.pm

```diff
@@ -537,6 +538,10 @@ sub dhcp
 {
     my ($self, $node, $st, $cmd, $dhcpmgr) = @_;
 
+    # If the profile has a discovery plugin configured, then don't second-guess
+    # it - it may or may not be ISC DHCP
+    return if $st->{configuration}->elementExists (DISCOVERY);
```

Comment:
add a $self->verbose("discovery plugin configured, no dhcp configuration") or something like that

Reply:
and the return condition should be that `$st->{configuration}->{enabled}` is true (so that one can disable it; and it should be disabled by default)

Reply:
I have no problem with adding a diag message, however, the code above is correct. The use of plugins is triggered by the existence of the plugin name - there does not have to be any specific keys under the name; see `iter_plugins()`. Neither `nbp` nor `osinstall` plugins are required to have a property called `enabled` - I don't see why `discovery` plugins should have that.

As I wrote above, `enabled` is really just a placeholder here - since the DHCP plugin does not have any plugin-wide options, it needs _something_ to force the path to exist.

Reply:
ok, if you really only need/want path existance, get rid of the enabled; the name suggests that you can also disable it.
you can just do 
```
discovery : dict = dict() with length(SELF) == 0
```
and get the same result.